File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.java
import java.util.*;

class Solution {
    private boolean[] visited;
    private Set<Integer> numbersSet;
    
    public int solution(String numbers) {
        visited = new boolean[numbers.length()];
        numbersSet = new HashSet<>();
        permutation(numbers.toCharArray(), "", numbers.length());
        
        int count = 0;
        for (int num : numbersSet) {
            if (isPrime(num)) {
                count++;
            }
        }
        return count;
    }
    
    private void permutation(char[] arr, String current, int len) {
        if (!current.isEmpty()) {
            numbersSet.add(Integer.parseInt(current));  // 정수로 중복 제거
        }
        for (int i = 0; i < len; i++) {
            if (!visited[i]) {
                visited[i] = true;
                permutation(arr, current + arr[i], len);
                visited[i] = false;
            }
        }
    }
    
    private boolean isPrime(int n) {
        if (n < 2) return false;
        if (n == 2 || n == 3) return true;
        if (n % 2 == 0 || n % 3 == 0) return false;
        for (int i = 5; i * i <= n; i += 6) {
            if (n % i == 0 || n % (i + 2) == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function solution(numbers) {
    const nums = numbers.split('');
    const numSet = new Set();
    
    function permute(current, used) {
        if (current.length > 0) {
            numSet.add(parseInt(current));  // 정수로 추가!
        }
        for (let i = 0; i < nums.length; i++) {
            if (!used[i]) {
                used[i] = true;
                permute(current + nums[i], used);
                used[i] = false;
            }
        }
    }
    
    permute('', new Array(nums.length).fill(false));
    
    let count = 0;
    numSet.forEach(num => {
        if (isPrime(num)) count++;
    });
    return count;
}

function isPrime(num) {
    if (num < 2) return false;
    if (num === 2 || num === 3) return true;
    if (num % 2 === 0 || num % 3 === 0) return false;
    for (let i = 5; i * i <= num; i += 6) {
        if (num % i === 0 || num % (i + 2) === 0) return false;
    }
    return true;
}
